pkg/server/plugin/keymanager/awskms/awskms.go

```diff
@@ -832,8 +836,8 @@ func parseAndValidateConfig(c string) (*Config, error) {
 		return nil, status.Error(codes.InvalidArgument, "configuration is missing a region")
 	}
 
-	if config.KeyMetadataFile == "" {
-		return nil, status.Error(codes.InvalidArgument, "configuration is missing server id file path")
+	if config.KeyMetadataFile == "" && config.KeyMetadata == "" {
```

Comment:
We should also check for the case were both settings are configured, and disallow having both of them configured at the same time. It would be confusing if we allow that.

Reply:
I think that we should have some validation of the `KeyMetadata` value that was configured. Values containing [characters that are not valid for a KMS Alias name](https://docs.aws.amazon.com/kms/latest/APIReference/API_CreateAlias.html#API_CreateAlias_RequestSyntax) will result in errors. I think that ensuring well-formed aliases _before_ trying to create them is important because the creation of the alias happens after the creation of the key, so users having a `KeyMetada` value with invalid characters will end up having an orphan key (until is disposed through the plugin routine to dispose stale keys) if the alias cannot be created.

Reply:
Do you know what's valid for Azure? GCP is a UUID which is already validated by the current code.

Reply:
Added checks for AWS.